File: reverseWordsInAString.py
class Solution:
    # @param s, a string
    # @return a string
    def reverseWords(self, s):
        s = s.strip()
        if not s:
        	return ""
        s = s.split()
        s.reverse()
        return " ".join(s)

if __name__ == '__main__':
	s = Solution()
	print s.reverseWords("a b")

File: Algorithms/reverseWordsInAString.java
public class Solution {
    public String reverseWords(String s) {
        s = s.trim();
        if (s.equals(""))
            return "";
        String[] t = s.split(" +");
        s = t[t.length-1];
        for (int i = t.length-2; i >= 0; i--) {
            s = s + " " + t[i];
        }
        return s;
    }
}
